cpp/unit/attribute.h

```diff
@@ -84,6 +91,8 @@ class AttributeContainer {
 		type{t} {}
 
 	attr_type type;
+
+	virtual AttributeContainer *copy() const = 0;
```

Comment:
Why are we even using this complicated `AttributeContainer`/`attr_type` model instead of simply making attributes polymorphic?